### PR TITLE
Update PagerState.currentPage when it is changed in LazyListState

### LINKED ISSUE #808: [Pager] CurrentPage is wrong after dataset change
## Describe the bug
If my dataset consists of A, B, C and I tap a button on page B which re-orders the dataset into B, A, C; I will continue seeing B and scrolling right shows A as expected. However the currentPage will still point to 1, immediately upon touching/scrolling the pager, it will update to 0. This happens using a HorizontalPager, although I would guess that the same thing applies for VerticalPager.

## To Reproduce
I think the description makes this clear, let me know if you want any additional information! 

This is what my logs look like. A SideEffect logs currentPage, and a snapshotFlow captures and logs the selected page whenever it changes.

(dataset is A,B,C)
Page: 0
Selected 1
Page: 1
(dataset changes to B,A,C)
Page: 1
Page: 1
(when I start scrolling the pager, I immediately see this call)
Selected 0 

## Expected behavior
The currentPage should be 0 since thats the page Im looking at.

## Environment:
 - Android OS version: API 30
 - Device: Emulator, Pixel 3a
 - Accompanist version: 0.20.0

## Additional context
A temporary workaround is to scroll to the selected page whenever it changes.

    LaunchedEffect(currentPage) {
        if (currentPage != state.targetPage) {
            state.animateScrollToPage(currentPage)
        }
    }

### PR DESCRIPTION
**Pull request by andkulikov**
It could happen for example if the items were reordered.

Fixes #808

### CODE REVIEW

File: pager/src/sharedTest/kotlin/com/google/accompanist/pager/BaseHorizontalPagerTest.kt [hunk 1]
```diff
@@ -132,17 +134,24 @@ abstract class BaseHorizontalPagerTest(
...
-                    modifier = Modifier.fillMaxSize()
+                    modifier = Modifier.fillMaxSize(),
+                    key = if (useKeys) {
+                        { pageToItem(it) }
+                    } else {
+                        null
+                    }
                 ) { page ->
+                    val item = pageToItem(page)
+                    println("asdasd item=$item page=$page")
```
Comment (chrisbanes) [inline pager/src/sharedTest/kotlin/com/google/accompanist/pager/BaseHorizontalPagerTest.kt:145]: Remove?

File: pager/src/sharedTest/kotlin/com/google/accompanist/pager/PagerTest.kt [hunk 2]
```diff
@@ -400,6 +401,32 @@ abstract class PagerTest {
...
+                pagerState = rememberPagerState().also { pagerState = it },
+                pageToItem = { list[it] },
+                useKeys = true
+            )
+        }
+
+        composeTestRule.runOnIdle {
+            assertThat(pagerState!!.currentPage).isEqualTo(0)
+            list = listOf("1", "0", "2")
+            println("asdasd change list")
```
Comment (chrisbanes) [inline pager/src/sharedTest/kotlin/com/google/accompanist/pager/PagerTest.kt:421]: Remove?

File: pager/src/sharedTest/kotlin/com/google/accompanist/pager/PagerTest.kt [hunk 3]
```diff
@@ -400,6 +401,32 @@ abstract class PagerTest {
...
+        }
+
+        composeTestRule.runOnIdle {
+            println("asdasd assert page")
```
Comment (chrisbanes) [inline pager/src/sharedTest/kotlin/com/google/accompanist/pager/PagerTest.kt:425]: Remove

**Review by chrisbanes — APPROVED** [review]
LGTM overall, just a few logs to remove

File: pager/src/sharedTest/kotlin/com/google/accompanist/pager/BaseHorizontalPagerTest.kt (hunk 1 above)
Comment (andkulikov) [reply pager/src/sharedTest/kotlin/com/google/accompanist/pager/BaseHorizontalPagerTest.kt]: Oops, thanks for noticing

**Review by JoseAlcerreca — APPROVED** [review]

**Pull request merged**